### DISCUSSION TITLE
run freertos after hss

### DISCUSSION THREAD

**Discussion by Nileena00** [polarfire-soc/polarfire-soc-documentation#465] [How to]
if i need to run a freetos on board after hss payload, may i know the procedures

**Comment by hughbreslin** [comment]
You will need to build FreeRTOS as a HSS payload (heres a [video](https://www.youtube.com/watch?v=LseJta8qVbM) outlining how things work) and program that to an SD card, eMMC, QSPI etc. You can follow the "Programming a Linux image" steps [here](https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/reference-designs-fpga-and-development-kits/updating-mpfs-kit.md#updating) (they're really how to program a HSS payload). There are some examples of running freeRTOS as a payload with the AMP documentation [here](https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/applications-and-demos/asymmetric-multiprocessing/amp.md#running-the-linux--freertosbm-amp-demo-on-the-icicle-kit) - there are prebuilt images you can use in the yocto releases to run the demos to save building them but you can also use the build set up as a reference.

**Reply by Nileena00** [reply]
thankyou so much!
